### PR TITLE
[rust-reqwest]: allow rust configuration struct to be deserialized

### PR DESCRIPTION
**Pull request by maxnachlinger**
Allow rust configuration struct to be deserialized, so it may be easily constructed from a file or other source.
Fixes: https://github.com/OpenAPITools/openapi-generator/issues/18687

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [ ] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.1.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [ ] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/rust/reqwest/configuration.mustache [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
 use http;
 use secrecy::{SecretString, ExposeSecret};
 {{/withAWSV4Signature}}
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, Deserialize)]
 pub struct Configuration {
     pub base_path: String,
     pub user_agent: Option<String>,
+    #[serde(skip)]
```
Comment (maxnachlinger) [inline modules/openapi-generator/src/main/resources/rust/reqwest/configuration.mustache:15]: We can't deserialize a reqwest client though :)

File: modules/openapi-generator/src/main/resources/rust/reqwest/configuration.mustache [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
+use serde::Deserialize;
 {{#withAWSV4Signature}}
 use std::time::SystemTime;
 use aws_sigv4::http_request::{sign, SigningSettings, SigningParams, SignableRequest};
 use http;
 use secrecy::{SecretString, ExposeSecret};
 {{/withAWSV4Signature}}
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, Deserialize)]
```
Comment (maxnachlinger) [inline modules/openapi-generator/src/main/resources/rust/reqwest/configuration.mustache:11]: This allows users to read configuration from a file, which I imagine is a common use case.

File: modules/openapi-generator/src/main/resources/rust/reqwest/configuration.mustache [hunk 3]
```diff
@@ -1,12 +1,24 @@
...
+pub struct DeserializedConfiguration {
+    pub base_path: String,
+    pub user_agent: Option<String>,
+    pub basic_auth: Option<BasicAuth>,
+    pub oauth_access_token: Option<String>,
+    pub bearer_access_token: Option<String>,
+    pub api_key: Option<ApiKey>,
+    // TODO: take an oauth2 token source, similar to the go one
+}
```
Comment (maxnachlinger) [inline modules/openapi-generator/src/main/resources/rust/reqwest/configuration.mustache:20]: This allows users to read configuration from a file, which I imagine is a common use case. 

Some of the structs used in config don't implement Deserialize - thinking of `Secret` for example. The properties of those types cannot be decorated with `#[serde(skip)]`  either as they don't implement `Default`. 

So I guess this struct is the best we can do with those constraints.

File: modules/openapi-generator/src/main/resources/rust/reqwest/configuration.mustache [hunk 4]
```diff
@@ -89,3 +101,18 @@ impl Default for Configuration {
         }
     }
 }
+
+impl From<DeserializedConfiguration> for Configuration {
```
Comment (maxnachlinger) [inline modules/openapi-generator/src/main/resources/rust/reqwest/configuration.mustache:105]: Make a `Configuration` from a `DeserializedConfiguration`

**Comment by wing328** [comment]
thanks for the PR
@frol (2017/07) @farcaller (2017/08) @richardwhiuk (2019/07) @paladinzh (2020/05) @jacob-pro (2022/10)

**Comment by wing328** [comment]
@maxnachlinger when you've time. can you please PM me via Slack to further discuss your Rust-related PRs? Thanks.

https://join.slack.com/t/openapi-generator/shared_invite/zt-12jxxd7p2-XUeQM~4pzsU9x~eGLQqX2g

**Pull request closed without merging**